src/documentation.ts

```diff
@@ -81,19 +77,22 @@ export class Documentation extends OpenApiBuilder {
   protected lastSecuritySchemaIds = new Map<SecuritySchemeType, number>();
   protected lastOperationIdSuffixes = new Map<string, number>();
   protected responseVariants = keys(defaultStatusCodes);
+  protected references = new Map<z.ZodTypeAny, ReferenceObject>();
 
   protected makeRef(
-    name: string,
-    schema: SchemaObject | ReferenceObject,
+    schema: z.ZodTypeAny,
+    depicted: SchemaObject | ReferenceObject,
+    name = this.references.get(schema)?.$ref.split("/").pop() ??
```

Comment:
should store just name, `make` should call `get`, and `get` should make an object with that name

Reply:
fixed